.github/actions/coverage-gate/check-per-file-coverage.py

```
@@ -0,0 +1,118 @@
+                    continue
+
+                fname = normalize_filename(raw_fname)
+                if fname.startswith("Zipper.Tests/") or fname.startswith("Zipper.Analyzers") or "/obj/" in fname or fname.startswith("obj/"):
+                    continue
+
+                lines = cls.findall("lines/line")
+                if not lines:
+                    continue
+
+                valid = len(lines)
+                covered = sum(1 for l in lines if int(l.get("hits", 0)) > 0)
```

Ruff (0.15.21)

[error] Ambiguous variable name: `l`

(E741)